common/src/main/java/org/corfudb/common/protocol/API.java

```diff
@@ -47,4 +124,163 @@ public static Request newPingRequest(Header header) {
+                .setHeader(header)
+                .setError(error)
+                .setPingResponse(pingResponse)
+                .build();
+    }
+
+    public static Request newRestartRequest(Header header) {
+        RestartRequest restartRequest = RestartRequest.getDefaultInstance();
+        return Request.newBuilder()
+                .setHeader(header)
+                .setRestartRequest(restartRequest)
+                .build();
+    }
+
+    public static Response newRestartResponse(Header header) {
+        return newRestartResponse(header, newNoServerError());
+    }
+
+    public static Response newRestartResponse(Header header, ServerError error) {
+        RestartResponse restartResponse = RestartResponse.getDefaultInstance();
+        return Response.newBuilder()
+                .setHeader(header)
+                .setError(error)
+                .setRestartResponse(restartResponse)
+                .build();
+    }
+
+    public static Request newResetRequest(Header header) {
+        ResetRequest resetRequest = ResetRequest.getDefaultInstance();
+        return Request.newBuilder()
+                .setHeader(header)
+                .setResetRequest(resetRequest)
+                .build();
+    }
+
+    public static Response newResetResponse(Header header) {
+        return newResetResponse(header, newNoServerError());
+    }
+
+    public static Response newResetResponse(Header header, ServerError error) {
+        ResetResponse resetResponse = ResetResponse.getDefaultInstance();
+        return Response.newBuilder()
+                .setHeader(header)
+                .setError(error)
+                .setResetResponse(resetResponse)
+                .build();
+    }
+
+    public static Request newSealRequest(Header header, long newEpoch) {
+        SealRequest sealRequest = SealRequest.newBuilder()
+                                        .setEpoch(newEpoch)
+                                        .build();
+        return Request.newBuilder()
+                .setHeader(header)
+                .setSealRequest(sealRequest)
+                .build();
+    }
+
+    public static Response newSealResponse(Header header) {
+        return newSealResponse(header, newNoServerError());
+    }
+
+    public static Response newSealResponse(Header header, ServerError error) {
+        SealResponse sealResponse = SealResponse.getDefaultInstance();
+        return Response.newBuilder()
+                .setHeader(header)
+                .setError(error)
+                .setSealResponse(sealResponse)
+                .build();
+    }
+
+    public static Response newErrorResponseNoPayload(Header header, ServerError error) {
+        return Response.newBuilder()
+                .setHeader(header)
+                .setError(error)
+                .build();
+    }
+
+    public static boolean validateRequest(Request request) {
+        switch(request.getHeader().getType()) {
+            case PING:
```

Comment:
This seems superfluous, just let the message go through and if not recognized the server should reply with an error.

Reply:
Can you comment a bit more on why this is superfluous? Currently, ResponseHandler/PeerClient performs this check as well using a similar switch statement. Also, I did have this check (as a checkArgument call) in the handle method for each message type (say handlePing when receiving a ping message), but you had mentioned that this was checked prior.
Although no error is currently sent, I do intend to send an error (see NettyServerRouter) if this checks fails.